### PR TITLE
Don't hardcode the function representation of builtin "once". Make emitLazyGlobalInitializer retrieve the convention from getBuiltinValueDecl.

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

File: lib/SILGen/SILGen.cpp [hunk 1]
```diff
@@ -975,10 +975,12 @@ SILFunction *SILGenModule::emitLazyGlobalInitializer(StringRef funcName,
                                                  PatternBindingDecl *binding,
                                                      unsigned pbdEntry) {
   ASTContext &C = M.getASTContext();
-  Type initType = FunctionType::get(
-                    TupleType::getEmpty(C), TupleType::getEmpty(C),
-                    FunctionType::ExtInfo()
-                      .withRepresentation(FunctionType::Representation::Thin));
+  auto *onceBuiltin =
+      dyn_cast<FuncDecl>(getBuiltinValueDecl(C, C.getIdentifier("once")));
```
Comment (rjmccall) [inline lib/SILGen/SILGen.cpp]: This should just be a ``cast<>``.

File: lib/SILGen/SILGen.cpp [hunk 2]
```diff
@@ -975,10 +975,12 @@ SILFunction *SILGenModule::emitLazyGlobalInitializer(StringRef funcName,
...
+  auto blockParam = onceBuiltin->getParameterLists()[0]->get(1);
+  auto *type = blockParam->getType()->getAs<FunctionType>();
```
Comment (rjmccall) [inline lib/SILGen/SILGen.cpp]: Similarly, this should be ``castTo<>``.

**Comment by rjmccall** [comment]
On the whole, looks fine.  Just a couple of minor requests to use asserting casts in places where you assume the cast isn't going to fail anyway.

**Comment by rjmccall** [comment]
Also, thank you for catching this oversight; I should've figured that we weren't really type-checking the SIL builtin instruction in the SIL verifier.

**Comment by kubamracek** [comment]
@swift-ci please test and merge

**Comment by kubamracek** [comment]
@swift-ci please test and merge

**Pull request merged**